src/annotations/AnnotationManager.php

```diff
@@ -322,6 +346,25 @@ protected function getAnnotations($class_name, $member_type = self::MEMBER_CLASS
     }
 
     /**
+     * Determines whether a class or trait has the specified member.
+     *
+     * @param string $className The name of the class or trait to check
+     * @param string $memberType The type of member, e.g. "property" or "method"
+     * @param string $memberName The member name, e.g. "method" or "$property"
+     *
+     * @return bool whether class or trait has the specified member
+     */
+    protected function classHasMember($className, $memberType, $memberName)
+    {
+        if ($memberType === self::MEMBER_METHOD) {
+            return method_exists($className, $memberName);
+        } else if ($memberType === self::MEMBER_PROPERTY) {
+            return property_exists($className, ltrim($memberName, '$'));
+        }
+        return false;
+    }
```

Comment:
Have you tried to `return null;`? Does anything break?

Yeah, I know that there are a lot of code, that looks like `foreach($this->getAnnotations(...) ...)` in AnnotationManager, but I hope for that code there never will be a situation when `null` is returned and `foreach` breaks.

Luckily publicly facing methods, like `getMethodAnnotations` and `getPropertyAnnotations` have own check for method/property existence, so don't worry for them.